### PR TITLE
[2.7][DependencyInjection] improve deprecation messages

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Also include the service id in the deprecation message.

### CODE REVIEW

**Comment by dosten** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 1]
```diff
@@ -151,7 +151,7 @@ private function parseDefinition(\DOMElement $service, $file)
         foreach (array('class', 'scope', 'public', 'factory-class', 'factory-method', 'factory-service', 'synthetic', 'lazy', 'abstract') as $key) {
             if ($value = $service->getAttribute($key)) {
                 if (in_array($key, array('factory-class', 'factory-method', 'factory-service'))) {
-                    @trigger_error(sprintf('The "%s" attribute in file "%s" is deprecated since version 2.6 and will be removed in 3.0. Use the "factory" element instead.', $key, $file), E_USER_DEPRECATED);
+                    @trigger_error(sprintf('The "%s" attribute of service "%s" in file "%s" is deprecated since version 2.6 and will be removed in 3.0. Use the "factory" element instead.', $key, (string) $service->getAttribute('id'), $file), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: the cast to string is not required, is it?

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**